cloudify_terminal/tasks.py

```diff
@@ -74,14 +78,20 @@ def run(**kwargs):
                                         log_file_name=log_file_name)
             ctx.logger.info("Will be used: " + ip)
             break
+
+        except socket.timeout as err:
+            ctx.logger.info("Socket timeout retrying... {}"
+                            .format(str(err)))
+
+        except ssh_exception.NoValidConnectionsError as err:
+            ctx.logger.info("NoValidConnectionsError retrying... {}"
+                            .format(str(err)))
+
         except Exception as ex:
-            ctx.logger.info("Can't connect to %s with %s" % (
-                repr(ip), str(ex)
-            ))
+            ctx.logger.info("Can't connect to:{} with exception:{} and type:{}"
+                            .format(repr(ip), str(ex), str(type(ex))))
     else:
-        raise cfy_exc.NonRecoverableError(
```

Comment:
I do not like that we're removing NonRecoverable error. We may spot some errors which are not captured properly and deployment may get stuck. I prefer to leave NonRecoverableError as a safety net ....but we need to properly report which error and type caused it.

Reply:
We need in someway separate "recoverable" errors, like network link will be restored until next retry and case that we never will have ip (from list) with alive connection. So we can think "all troubles is temporary, lets try one more time"  for now.

Reply:
It may work :) Let me test it.